### ISSUE TITLE
Persist raw data from kafka topic as it is

### ISSUE THREAD

**Issue by saumya2700** [apache/pinot#8399]
**Feature request:** 
To save raw data from topic in pinot table.

**Use case :** 
We have lots of complex schemas and we are using pinot for saving and retreiving topic data with times stamp and some other fields.
We do not want to map all nested columns from complex schema and create pinot schema and use lots of transformation functions. There are some places we want raw data as it is in pinot table. 

**Sample data :** 

`{
    "header": {
        "tid": "12wee",
        "rid": 1,
        "timestamp": 1647347092337
    },
    "status": "200_SUCCESS",
    "jasData": {
        "sdata": -22.89122,
        "cnn": 0.823469,
        "kli": 2.238848,
        "olp": [
            {
                "ovPerc": 0.032486767,
                "hg": 30.0,
                "abshi": 6.661863
            }
        ],
        "terrkl": {
            "ovPerc": 0.9675132,
            "dist": [
                -25.17232,
                -25.17232,
                -25.130081
            ]
        },
        "bcut": 2.77
    },
    "rgData": {
        "pre": 102033.33,
        "pv": 0.16,
        "t": 287.36,
        "timestamp": 1647347069000
    },
        "timestamp": 1647347092337
    }
}`

**Comment by Jackie-Jiang** [comment]
Do you want the whole record to be stored as a json field, or a field to be stored as a json field? Pinot can achieve the latter by the `jsonFormat` ingestion transform function (e.g. `jsonFormat(jasData)` to store the `jasData` as a json). Storing whole record as json is not supported currently.

**Comment by saumya2700** [comment]
@Jackie-Jiang we want whole record to be stored as json or bytes. Latter one with jsonformat is working but our use case is to store whole message.

**Comment by Jackie-Jiang** [comment]
We may consider adding a new config in the `IngestionConfig` to store the json string of the record into a field. The logic needs to be implemented into the `RecordExtractor`

**Comment by suddendust** [comment]
@Jackie-Jiang Should this not be done for any input type and not just JSON?

**Comment by Jackie-Jiang** [comment]
@suddendust Correct. The `RecordExtractor` for each input type should support this. The raw data can be stored as JSON string though. We might not want to store different type raw data for different input type because that might not be readable/queryable. E.g. raw bytes for Avro records

**Comment by suddendust** [comment]
@Jackie-Jiang This sounds good to me. @saumya2700 If you're not working on this, I can pick this up?

**Comment by saumya2700** [comment]
> @Jackie-Jiang This sounds good to me. @saumya2700 If you're not working on this, I can pick this up?

yes Please go ahead.

**Comment by suddendust** [comment]
@Jackie-Jiang So I gave this a thought. Even if this new property is set to true, then filter configs and transform configs should keep working, right? As in, Pinot should store these records as JSON after applying the filters and the transforms. Or should be store the record as-is? I think if the user has provided the filter or the transform configs, we should apply then. Else, store as-is. What do you think?

**Comment by Jackie-Jiang** [comment]
> @Jackie-Jiang If this property is set to true, then we should just ignore the filter and transform configs? Or should be provide a way to store the record as a JSON AFTER applying the filters and the transforms? IMO, most users would just want to store their records as-is, so we should just ignore any transform or filter configs.

@suddendust IMO we should still honor the filter and transform configs. This new config means we want to store the whole record into a configured field, but we should still support filtering/extracting fields from the records. User might choose to use this new config to capture the whole record, but still extract the fields that are commonly used in the query for performance reason.
